api/ipfsproxy/config.go

```diff
@@ -0,0 +1,180 @@
+package ipfsproxy
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	ma "github.com/multiformats/go-multiaddr"
+
+	"github.com/ipfs/ipfs-cluster/config"
+)
+
+const configKey = "ipfsproxy"
+
+// Default values for Config.
+const (
+	DefaultProxyAddr              = "/ip4/127.0.0.1/tcp/9095"
+	DefaultNodeAddr               = "/ip4/127.0.0.1/tcp/5001"
+	DefaultProxyReadTimeout       = 0
+	DefaultProxyReadHeaderTimeout = 5 * time.Second
+	DefaultProxyWriteTimeout      = 0
+	DefaultProxyIdleTimeout       = 60 * time.Second
+	DefaultIPFSRequestTimeout     = 5 * time.Minute
```

Comment:
It is used here https://github.com/ipfs/ipfs-cluster/pull/576/files#diff-40d1514e01fa463b9839f0d7c6625184R398

Reply:
Let's use the `r.Context()` instead of a timeout. You can create the contexts using: https://github.com/ipfs/ipfs-cluster/blob/master/rpcutil/rpcutil.go#L35